### ISSUE TITLE
Disabling a non-existant check should not be a fatal error

### ISSUE THREAD

**Issue by natefinch** [go-critic/go-critic#1398]
At work, we had sloppyTestFuncName set as disabled, because it found false positives.

Then y'all removed it.  ...but the problem is that this now breaks our configuration.
The configuration that used to be valid, now produces an error. Even though there is no actual change in what checks are run between the old version of gocritic and the new.

This breaks backwards compatibility with existing configurations for basically no reason.

The config tells you not to run a check you don't recognize. Well, you weren't gonna run that anyway, so, all done, I guess?

Sure, print a warning message that a disabled check doesn't actually exist, but don't stop running the whole thing when you *know* there won't be any change to behavior.

**Comment by cristaloleg** [comment]
Yeah, kinda make sense. Sorry for that. Wanna make a PR ? 😉

**Comment by quasilyte** [comment]
warnings tend to get ignored and some tools end up having 10+ warnings printed out for no reason.

I like the idea that requesting something that does not exist is an error. We can disagree on that one if you would like.

For me, the main problem is the checker being removed. We usually don't do that.

And if we do, it's usually a checker that was marked with an experimental tag.
If users rely on experimental checkers stay forever - it's outside of the backward compatibility promises.

https://github.com/go-critic/go-critic/commit/404c56b4015addd8f714e71c4204158e5773607f#diff-389d96f065f3108bd8fbfd7a928e9c9b81f4f5f308a021fc539fc09a49345763L789

The checker you've mentioned was experimental.
If think it's fair to say that experimental stuff can break.

Unless we remove the experimental tag, the checker is basically in its public testing phase.
You may use it if you find it useful, but we still have some freedom in making changes and/or removing it.

**Comment by ehsundar** [comment]
How can I reproduce this fault @natefinch?

```bash
$ ./bin/gocritic check -disable 'ehsandar,offBy1,dummyName,#dummyTag' -enable 'offBy1' ./...
[offBy1]
[ehsandar offBy1 dummyName #dummyTag]
init checkers: empty checkers set selected
```

**Comment by natefinch** [comment]
I was running gocritic via golangci-lint, so I am not sure how to do it via gocritic itself, but I just had `sloppyTestFuncName` configured to be disabled. Ironically, I realize now that we had also removed experimental checkers from our list of enabled categories, so it was redundant to have it in the disabled list.
